extension/test/cleanup-ctp-project.js

```diff
@@ -4,8 +4,9 @@ const config = require('../src/config/config')
 
 before(async () => {
   const ctpProjectKeys = config.getAllCtpProjectKeys()
-  for (const projectKey of ctpProjectKeys) {
-    const ctpClient = ctpClientBuilder.get(projectKey)
+  for (const ctpProjectKey of ctpProjectKeys) {
+    const ctpConfig = config.getCtpConfig(ctpProjectKey)
```

Comment:
I did not get those cascading changes on every test, I guess it was better before in extension because when you know the project key you know the ctp config ? 🤔 Am I missing something ?

Reply:
Yes, that's correct. Now there could be multiple ctp projects, so I made this kind of clean up.

Reply:
So I'm not missing anything, and it was cleaner before ?

This part is added everywhere on client creation on tests : `const ctpConfig = config.getCtpConfig(ctpProjectKey)` 
this was in ctpClient get function before. imo you could get all config with only knowing ctpProjectKey (before it was that way).

Anyways you could keep it if you see a benefit.

Reply:
Yes, you need to know the project key now to get the config. Before it was not necessary as you have only one config.

Reply:
I feel these discussions will not end 🚀 

Why you are not passing only the projectKey then, because you could get the ctpConfig inside the ctp.js ? Also it's not in this branch that we are passing projectKey to the ctp.js.

Reply:
I'm not passing just projectKey because I felt like the client should not load config, but rather it should get the config from outside. It's also easier for testing if we can pass modified config.